crates/xxi-node/src/commons/funding_fee_event.rs

```diff
@@ -0,0 +1,67 @@
+use crate::commons::to_nearest_hour_in_the_past;
+use crate::commons::ContractSymbol;
+use crate::commons::Direction;
+use bitcoin::SignedAmount;
+use rust_decimal::Decimal;
+use serde::Deserialize;
+use serde::Serialize;
+use time::OffsetDateTime;
+
+/// The funding rate for any position opened before the `end_date`, which remained open through the
+/// `end_date`.
+#[derive(Serialize, Clone, Copy, Deserialize, Debug)]
+pub struct FundingRate {
+    /// A positive funding rate indicates that longs pay shorts; a negative funding rate indicates
+    /// that shorts pay longs.
+    rate: Decimal,
+    /// The start date for the funding rate period. This value is only used for informational
+    /// purposes.
+    ///
+    /// The `start_date` is always a whole hour.
+    start_date: OffsetDateTime,
```

Comment:
🤔 do we really need a start date? isn't the start date the last end_date?

Reply:
The start date is the last end date if everything goes to plan. It's not strictly necessary, but it means that if you load a single row you have all the useful context.